cpu/native/include/socket_zep.h

```diff
@@ -58,50 +58,54 @@
 
 #include "net/netdev.h"
 #include "net/netdev/ieee802154.h"
+#include "net/ieee802154/radio.h"
 #include "net/zep.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief   ZEP device initialization parameters
+ */
+typedef struct {
+    char *local_addr;   /**< local address string */
+    char *local_port;   /**< local address string */
+    char *remote_addr;  /**< remote address string */
+    char *remote_port;  /**< local address string */
+} socket_zep_params_t;
+
 /**
  * @brief   ZEP device state
  */
 typedef struct {
-    netdev_ieee802154_t netdev;     /**< netdev internal member */
+    const socket_zep_params_t *params;
+    ieee802154_dev_t *hal;          /**< radio HAL */
```

Comment:
in the SPI radios it's possible to pass this pointer as an ISR argument, so you don't need to store it here. Would it be possible here?

Reply:
For the ISR this works, but there are other places where I need to access `sock_fd`

Reply:
but why does the private descriptor needs access to the hal? can't this be accessed directly from the private descriptor?